examples/DataApi-NFT/template/modules/shared_data/src/main/scala/com/my/nft_example/shared_data/Combiners.scala

```diff
@@ -0,0 +1,57 @@
+package com.my.nft_example.shared_data
+
+import com.my.nft_example.shared_data.Data.{Collection, MintCollection, MintNFT, NFT, State, TransferCollection, TransferNFT}
+import com.my.nft_example.shared_data.Utils.customUpdateSerialization
+import monocle.Monocle.toAppliedFocusOps
+import org.tessellation.schema.address.Address
+import org.tessellation.security.hash.Hash
+
+import java.util.Calendar
+import scala.collection.mutable
+
+object Combiners {
+  def combineMintCollection(update: MintCollection, acc: State, collectionOwner: Address): State = {
+    val collectionId = Hash.fromBytes(customUpdateSerialization(update)).toString
+    val nowInTime = Calendar.getInstance().getTime.getTime
+    val newState = Collection(collectionId, collectionOwner, update.name, nowInTime, Map.empty)
+
+    acc.focus(_.collections).modify(_.updated(collectionId, newState))
+  }
+
+  def combineMintNFT(update: MintNFT, acc: State): State = {
+    val nowInTime = Calendar.getInstance().getTime.getTime
+    val newNFT = NFT(update.nftId, update.collectionId, update.owner, update.uri, update.name, update.description, nowInTime, update.metadata)
+
+    val collection = acc.collections(update.collectionId)
+    val collectionNFTs = collection.nfts ++ Map(update.nftId -> newNFT)
```

Comment:
` val value = Map(1 -> 1, 2 -> 2, 3 -> 3) + (4 -> 4)`
Example for single addition. Yields:

`map: Map(1 -> 1, 2 -> 2, 3 -> 3, 4 -> 4)`